### ISSUE TITLE
Xwayland error

### ISSUE THREAD

**Issue by RaWolFoX** [termux/termux-x11#284]
### Problem description

```
Starting Xwayland                                 error in client communication (pid 16381)
(EE)                                              Fatal server error:
(EE) wl_display@1: error 0: invalid object 11     (EE)
Xwayland died, no need to go on...                Aborted
```

### What steps will reproduce the bug?

```
termux-x11 :0
```

### System information

termux-info:

<details>

```
Termux Variables:
TERMUX_API_APP__VERSION_NAME=0.50.1+d2f2051
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP__AM_SOCKET_SERVER_ENABLED=true
TERMUX_APP__APK_PATH=/data/app/~~ZgyJQXHPTG0WsX8DfRcxpw==/com.termux-AK32nZyPfShLqH6m0lb-dA==/base.apk
TERMUX_APP__APK_RELEASE=GITHUB
TERMUX_APP__FILES_DIR=/data/user/0/com.termux/files
TERMUX_APP__IS_DEBUGGABLE_BUILD=true
TERMUX_APP__IS_INSTALLED_ON_EXTERNAL_STORAGE=false
TERMUX_APP__PACKAGE_MANAGER=apt
TERMUX_APP__PACKAGE_NAME=com.termux
TERMUX_APP__PACKAGE_VARIANT=apt-android-7
TERMUX_APP__PID=15251
TERMUX_APP__SE_FILE_CONTEXT=u:object_r:app_data_file:s0:c184,c258,c512,c768
TERMUX_APP__SE_INFO=default:targetSdkVersion=28:complete
TERMUX_APP__SE_PROCESS_CONTEXT=u:r:untrusted_app_27:s0:c184,c258,c512,c768
TERMUX_APP__TARGET_SDK=28
TERMUX_APP__UID=10696
TERMUX_APP__USER_ID=0
TERMUX_APP__VERSION_CODE=118
TERMUX_APP__VERSION_NAME=0.118.0+9b274f9
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0+9b274f9
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://packages.termux.dev/apt/termux-main stable main
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.dev/apt/termux-x11/ x11 main
Updatable packages:
All packages up to date
termux-tools version:
1.37.0
Android version:
11
Kernel build information:
Linux localhost 4.14.180-perf+ termux/termux-packages#2 SMP PREEMPT Sat Sep 3 23:17:03 CST 2022 aarch64 Android
Device manufacturer:
OPPO
Device model:
CPH1937
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
```
</details>

**Comment by twaik** [comment]
Try `WAYLAND_DEBUG=1 termux-x11 :0`. I need to see debug log.

**Comment by RaWolFoX** [comment]
> Try `WAYLAND_DEBUG=1 termux-x11 :0`. I need to see debug log.

┌──[~]
└─➤ WAYLAND_DEBUG=1 termux-x11 :0
Starting Xwayland
[2994359.775]  -> wl_display@1.get_registry(new id
 wl_registry@2)
[2994359.892]  -> wl_display@1.sync(new id wl_call
back@3)
[2994360.661] wl_display@1.get_registry(new id wl_
registry@2)
[2994360.874]  -> wl_registry@2.global(1, "wl_comp
ositor", 5)
[2994360.928]  -> wl_registry@2.global(2, "wl_seat
", 8)
[2994360.954]  -> wl_registry@2.global(3, "wl_outp
ut", 4)
[2994360.979]  -> wl_registry@2.global(4, "wl_shel
l", 1)
[2994361.002]  -> wl_registry@2.global(5, "xdg_wm_
base", 5)
[2994361.024]  -> wl_registry@2.global(6, "wl_shm"
, 1)
[2994361.053] wl_display@1.sync(new id wl_callback
@3)
[2994361.076]  -> wl_callback@3.done(0)
[2994361.097]  -> wl_display@1.delete_id(3)
[2994361.336] wl_display@1.delete_id(3)
[2994361.422] wl_registry@2.global(1, "wl_composit
or", 5)
[2994361.457]  -> wl_registry@2.bind(1, "wl_compos
itor", 4, new id [unknown]@4)
[2994361.486] wl_registry@2.global(2, "wl_seat", 8
)
[2994361.505] wl_registry@2.global(3, "wl_output",
 4)
[2994361.526]  -> wl_registry@2.bind(3, "wl_output
", 4, new id [unknown]@5)
[2994361.631] wl_registry@2.global(4, "wl_shell", 
1)
[2994361.656] wl_registry@2.global(5, "xdg_wm_base
", 5)
[2994361.676]  -> wl_registry@2.bind(5, "xdg_wm_ba
se", 1, new id [unknown]@6)
[2994361.698] wl_registry@2.global(6, "wl_shm", 1)
[2994361.716]  -> wl_registry@2.bind(6, "wl_shm", 
1, new id [unknown]@7)
[2994361.741] wl_callback@3.done(0)
[2994362.293]  -> wl_display@1.sync(new id wl_call
back@3)
[2994365.936] wl_registry@2.bind(1, "wl_compositor
", 4, new id [unknown]@4)
[2994366.081] wl_registry@2.bind(3, "wl_output", 4
, new id [unknown]@5)
[2994366.165]  -> wl_output@5.geometry(0, 0, 211, 
158, 0, "Lorie", "none", 0)
[2994366.224]  -> wl_output@5.scale(1)
[2994366.258]  -> wl_output@5.mode(3, 800, 600, 60
000)
[2994366.291]  -> wl_output@5.done()
[2994366.340] wl_registry@2.bind(5, "xdg_wm_base",
 1, new id [unknown]@6)
[2994366.421] wl_registry@2.bind(6, "wl_shm", 1, n
ew id [unknown]@7)
[2994366.460]  -> wl_shm@7.format(0)
[2994366.488]  -> wl_shm@7.format(1)
[2994366.520] wl_display@1.sync(new id wl_callback
@3)
[2994366.552]  -> wl_callback@3.done(0)
[2994366.581]  -> wl_display@1.delete_id(3)
[2994366.812] wl_display@1.delete_id(3)
[2994366.851] wl_output@5.geometry(0, 0, 211, 158,
 0, "Lorie", "none", 0)
[2994366.876] wl_output@5.scale(1)
[2994366.888] wl_output@5.mode(3, 800, 600, 60000)
[2994366.902] wl_output@5.done()
[2994366.970] wl_callback@3.done(0)
[2994560.676]  -> wl_shm@7.create_pool(new id wl_s
hm_pool@3, fd 8, 1920000)
[2994560.793]  -> wl_shm_pool@3.create_buffer(new 
id wl_buffer@8, 0, 800, 600, 3200, 1)
[2994560.809]  -> wl_shm_pool@3.destroy()
[2994563.338]  -> wl_compositor@4.create_surface(n
ew id wl_surface@9)
[2994563.404]  -> xdg_wm_base@6.get_xdg_surface(ne
w id xdg_surface@10, wl_surface@9)
[2994563.418]  -> xdg_surface@10.get_toplevel(new 
id xdg_toplevel@11)
[2994563.431]  -> xdg_toplevel@11.set_title("Xwayl
and on :0")
[2994563.439]  -> xdg_toplevel@11.set_app_id("org.
freedesktop.Xwayland")
[2994563.448]  -> wl_surface@9.commit()
[2994563.455]  -> wl_compositor@4.create_region(ne
w id wl_region@12)
[2994563.464]  -> wl_region@12.add(0, 0, 800, 600)
[2994563.474]  -> wl_surface@9.set_opaque_region(w
l_region@12)
[2994563.482]  -> wl_region@12.destroy()
[2994563.691] wl_shm@7.create_pool(new id wl_shm_p
ool@3, fd 60, 1920000)
[2994563.776] wl_shm_pool@3.create_buffer(new id w
l_buffer@8, 0, 800, 600, 3200, 1)
[2994563.823] wl_shm_pool@3.destroy()
[2994563.843]  -> wl_display@1.delete_id(3)
[2994563.863] wl_compositor@4.create_surface(new i
d [unknown]@9)
[2994563.925] xdg_wm_base@6.get_xdg_surface(new id
 [unknown]@10, wl_surface@9)
[2994563.955] xdg_surface@10.get_toplevel(new id [
unknown]@11)
[2994563.975]  -> wl_display@1.error(wl_display@1,
 0, "invalid object 11")
error in client communication (pid 28277)
[2994660.690]  -> wl_shm@7.create_pool(new id wl_s
hm_pool@13, fd 8, 1024)
[2994660.813]  -> wl_shm_pool@13.create_buffer(new
 id wl_buffer@14, 0, 16, 16, 64, 0)
[2994660.829]  -> wl_shm_pool@13.destroy()
[2994662.724]  -> wl_display@1.get_registry(new id
 wl_registry@15)
[2994662.844]  -> wl_display@1.sync(new id wl_call
back@16)
[2994662.911] wl_display@1.delete_id(3)
[2994662.928] wl_display@1.error(wl_display@1, 0, 
"invalid object 11")
(EE)
Fatal server error:
(EE) wl_display@1: error 0: invalid object 11
(EE)
Xwayland died, no need to go on...
Aborted
┌──[~]
└─➤

**Comment by twaik** [comment]
What build did you use?

**Comment by RaWolFoX** [comment]
> What build did you use?

https://github.com/termux/termux-x11/actions/runs/4502337109

Latest github build.

**Comment by twaik** [comment]
Try this one. https://github.com/termux/termux-x11/actions/runs/4369839485

**Comment by twaik** [comment]
And please, post output of this command: `Xwayland -version`.

**Comment by RaWolFoX** [comment]
> And please, post output of this command: `Xwayland -version`.

The X.Org Foundation Xwayland Version 23.1.0 (12301000)
X Protocol Version 11, Revision 0

Still not working

**Comment by RaWolFoX** [comment]
After the xwayland update, something is broken🥲

**Comment by twaik** [comment]
I am not sure what exactly happens. I'll check this.

**Comment by RaWolFoX** [comment]
> I am not sure what exactly happens. I'll check this.

Ok .

**Comment by twaik** [comment]
Ok, I see the problem. I implemented fake/stub compositor and it ignores some requests. Newer Xwayland uses some requests I did not implement. I'll fix it soon.

**Comment by ask9027** [comment]
> Ok, I see the problem. I implemented fake/stub compositor and it ignores some requests. Newer Xwayland uses some requests I did not implement. I'll fix it soon.

@twaik did you fix it?
becouse i don't want to use VNC Viewer.😃

**Comment by twaik** [comment]
Fixed in https://github.com/termux/termux-x11/commit/b96c785513c1e28c3a8b3ebf5d7da0ff64ed91e0

**Comment by ask9027** [comment]
> Fixed in [b96c785](https://github.com/termux/termux-x11/commit/b96c785513c1e28c3a8b3ebf5d7da0ff64ed91e0)

You are awesome 😎.

**Comment by RaWolFoX** [comment]
> Fixed in https://github.com/termux/termux-x11/commit/b96c785513c1e28c3a8b3ebf5d7da0ff64ed91e0

😍😍😍😍
